dragonfly/opt/unittest_euclidean_random_optimiser.py

```diff
@@ -165,7 +164,21 @@ def run_optimiser(cls, func_caller, worker_manager, max_capital, mode, *args, **
     """ Runs optimiser. """
     return random_optimiser.random_optimiser_from_func_caller(
              func_caller, worker_manager, max_capital, mode)
-
+  
+  def test_ask_tell(self):
+    """ Testing random optimiser with ask tell interface. """
+    self.report('Testing %s using the ask-tell interface.'%(type(self)))
+    opt = random_optimiser.EuclideanRandomOptimiser(self.func_caller, self.worker_manager_1)
```

Comment:
@kirthevasank is it necessary that the optimizer is instantiated with `self.func_caller, self.worker_manager_1`? 

For an external library calling into Dragonfly, it would be nice to simply do the following:

```
opt = random_optimiser.EuclideanRandomOptimiser()
opt.ask()
opt.tell(x, y)
```

Reply:
A function_caller is not really necessary, but this is how ExperimentDesigner knows what the domain of the function is. We should figure out a way to pass the domain to the optimiser.
A worker_manager should not be necessary.